eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -141,6 +156,20 @@ public function buildDomainFields(array $spiFields, $contentType)
                 continue;
             }
 
+            if ($languages !== null && !in_array($spiField->languageCode, $languages)) {
+                // If filtering is enabled we ignore fields in other languages then $fieldLanguages, if:
+                if ($alwaysAvailableLanguage === null) {
+                    // Ignore field if we don't have $alwaysAvailableLanguageCode fallback
+                    continue;
+                } elseif (!empty($fieldInFilterLanguagesMap[$spiField->fieldDefinitionId])) {
+                    // Ignore field if it exists in one of the filtered languages
+                    continue;
+                } elseif ($spiField->languageCode !== $alwaysAvailableLanguage) {
+                    // Also ignore if field is not in $alwaysAvailableLanguageCode
+                    continue;
+                }
+            }
```

Comment:
> This could be simplified with a goto

There was a time where this would have been quoted, and people would have grinned.